CHANGELOG.md

```diff
@@ -9,6 +9,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Disallow multiple data paths for search nodes ([#6427](https://github.com/opensearch-project/OpenSearch/pull/6427))
 
 ### Dependencies
+- Bump `com.azure:azure-storage-common` from 12.19.3 to 12.20.0
```

Comment:
Similar to the comment on #6503 

>This will make the `CHANGELOG` section for `Unreleased 2.x` on `main` and `2.x` diverge.
We would also need to update the changelog on main post this PR merge or fix the changelog on main and re-run the backport workflow, which will take care of it automagically.